database/utils.go

```diff
@@ -43,6 +45,49 @@ func SplitOnDupId[T IDer]() com.BulkChunkSplitPolicy[T] {
 	}
 }
 
+// InsertObtainID executes the given query and fetches the last inserted ID.
+//
+// Using this method for database tables that don't define an auto-incrementing ID, or none at all,
+// will not work. The only supported column that can be retrieved with this method is id.
+// This function expects [TxOrDB] as an executor of the provided query, and is usually a *[sqlx.Tx] or *[DB] instance.
+// Returns the retrieved ID wrapped in [types.Int] on success and error on any database inserting/retrieving failure.
+func InsertObtainID(ctx context.Context, conn TxOrDB, stmt string, arg any) (types.Int, error) {
+	var resultID int64
+	switch conn.DriverName() {
+	case PostgreSQL:
+		query := stmt + " RETURNING id"
+		ps, err := conn.PrepareNamedContext(ctx, query)
+		if err != nil {
+			return types.Int{}, errors.Wrapf(err, "cannot prepare %q", query)
+		}
+		// We're deferring the ps#Close invocation here just to be on the safe side, otherwise it's
+		// closed manually later on and the error is handled gracefully (if any).
+		defer func() { _ = ps.Close() }()
+
+		if err = ps.GetContext(ctx, &resultID, arg); err != nil {
+			return types.Int{}, CantPerformQuery(err, query)
+		}
+
+		if err = ps.Close(); err != nil {
+			return types.Int{}, errors.Wrapf(err, "cannot close prepared statement %q", query)
+		}
+	case MySQL:
+		result, err := sqlx.NamedExecContext(ctx, conn, stmt, arg)
+		if err != nil {
+			return types.Int{}, CantPerformQuery(err, stmt)
+		}
+
+		resultID, err = result.LastInsertId()
+		if err != nil {
+			return types.Int{}, errors.Wrap(err, "cannot retrieve last inserted ID")
+		}
```

Comment:
I mean, you could also make this the default branch.

Reply:
And now this two-case switch could be an _if_.

Reply:
And why is that necessary? I think you are just focussing on the non-essential things.